### PR TITLE
Add checkstyle to Gradle build

### PR DESCRIPTION
**Pull request by Sieberkev**
**Fixed issue:** #1125

**Changes made:**

* Added checkstyle to build.gradle (only check "src/main/kotlin" and not the libraries)
* Added a checkstyle.xml I found in another JAVA/kotlin project
* Fixed several checkstyle errors (File contains tab characters, Space needed before opening parenthesis, Line has trailing spaces) so the current checkstyle.xml builds without errors

**To dos**

- [x] Make checkstyle work on `gradlew build`
- [ ] Implement the default IntelliJ checks
- [ ] Why does the unused imports/star imports check doesn't work?
- [ ] Imports in lexographical order
- [ ] Make sure all current files are corrected

### CODE REVIEW

File: checkstyle.xml [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    <property name="message" value="Space needed before opening parenthesis."/>
+  </module>
+
+  <!-- For each spacing -->
+  <module name="RegexpSingleline">
+    <property name="format" value="^\s*for \(.*?([^ ]:|:[^ ])"/>
+    <property name="message" value="Space needed around ':' character."/>
+  </module>
+
+  <module name="TreeWalker">
```
Comment (Sieberkev) [inline checkstyle.xml]: For some reason not a single TreeWalker submodule check works but there's no error whatsoever.

And my googlefu is not helping me :cry:

**Comment by Zizekftw** [comment]
TreeWalker submodules have been breaking as you advance JDK versions. At work, we've had to remove those checks from the XML file. DoubleCheckedLocking and RedundantThrows used to work with Java 6, but haven't worked with 7 or 8.

http://checkstyle.sourceforge.net/releasenotes.html#Release_6.2 (For removal of Redundant Throws)
http://stackoverflow.com/questions/12795935/checkstyle-eclipse-plugin-doesnt-work (For removal of DoubleCheckedLocking).

Checkstyle has a history of making non-backwards-compatible modifications.

**Comment by Sieberkev** [comment]
> # Release 7.0 # 
> Breaking backward compatibility:
>  * update checkstyle to jdk8 as runtime jdk

7.0 should work with Java JDK8, no?

**Comment by Zizekftw** [comment]
It should work. Our work hasn't upgraded from 6.14 yet as there were no new features that we needed so I can't comment on that specifically.

I just know at version 6.14 those TreeWalker checks weren't working. I think those features were removed from the Eclipse plugin so it would throw an "Unable to instantiate module <moduleName>"

FWIW: One other check we made that was useful was to ensure that no println statements were included so that everything committed had logging done via the application logger. Up to you guys if you think it's valuable, but here's the xml to include if you want it.

```
<module name="Regexp">
    <metadata name="net.sf.eclipsecs.core.comment" value="System.out.println statements are not allowed."/>
    <property name="format" value="System\.out\.println"/>
    <property name="illegalPattern" value="true"/>
    <property name="ignoreComments" value="true"/>
</module>
```

**Comment by Sieberkev** [comment]
Borked up my fork repository/branch, will resubmit

**Pull request closed without merging**